distributed/tests/test_cancelled_state.py

```diff
@@ -561,3 +567,129 @@ def test_resume_executing_worker_state(ws_with_running_task):
     assert not instructions
     assert ws.tasks["x"] is ts
     assert ts.state == prev_state
+
+
+def test_workerstate_flight_to_flight(ws):
+    """Test state loop:
+
+    flight -> cancelled -> fetch
+
+    Test that the task immediately reverts to its original state.
+    """
+    ws2 = "127.0.0.1:2"
+    instructions = ws.handle_stimulus(
+        ComputeTaskEvent.dummy("y", who_has={"x": [ws2]}, stimulus_id="s1"),
+        FreeKeysEvent(keys=["y", "x"], stimulus_id="s2"),
+        ComputeTaskEvent.dummy("z", who_has={"x": [ws2]}, stimulus_id="s3"),
+    )
+    assert instructions == [
+        GatherDep(worker=ws2, to_gather={"x"}, total_nbytes=1, stimulus_id="s1")
```

Comment:
Does it make sense to fire off a second GatherDep for `x` if the previous one is already ongoing?

Reply:
There is no second GatherDep in the instructions.
In flight tasks can be in flight from a single worker at a time.

Reply:
I overlooked that we're firing _all_ instructions in a single call to `ws.handle_stimulus`. I thought that this was just the last `ComputeTaskEvent` creating the instruction.